addons/mail/static/src/core/common/composer.js

```diff
@@ -515,18 +515,22 @@ export class Composer extends Component {
 
     async sendMessage() {
         await this.processMessage(async (value) => {
-            const postData = {
-                attachments: this.props.composer.attachments,
-                isNote: this.props.type === "note",
-                mentionedChannels: this.props.composer.mentionedChannels,
-                mentionedPartners: this.props.composer.mentionedPartners,
-                cannedResponseIds: this.props.composer.cannedResponses.map((c) => c.id),
-                parentId: this.props.messageToReplyTo?.message?.id,
-            };
+            const postData = this.postData();
             await this._sendMessage(value, postData);
         });
     }
 
+    postData() {
+        return {
+            attachments: this.props.composer.attachments,
+            isNote: this.props.type === "note",
+            mentionedChannels: this.props.composer.mentionedChannels,
+            mentionedPartners: this.props.composer.mentionedPartners,
+            cannedResponseIds: this.props.composer.cannedResponses.map((c) => c.id),
+            parentId: this.props.messageToReplyTo?.message?.id,
+        };
+    }
```

Comment:
Ok for easing implementation in this PR, but note that this increases quite significantly the technical debt of code, as this is another level of indirection and a new patch, which will require far more effort to clean and simplify.

We should be really cautious with such significant increase of code maintenance, because eventually it's gonna become too much effort to clean the code and we are stuck with crap code because no one will want to make the effort to clean the code.

As this code coverage is heavily related to the composer state, the function param passed to other functions should probably be simplified to just composer, rather than many transformed props originating from composer.